functional-tests/tests/prover/prover_el_selfdestruct_to_address.py

```diff
@@ -0,0 +1,88 @@
+import flexitest
+from solcx import install_solc, set_solc_version
+from web3 import Web3
+
+from envs import testenv
+from utils import (
+    el_slot_to_block_commitment,
+    wait_for_proof_with_time_out,
+    wait_until_with_value,
+)
+from utils.transaction import SmartContracts
+
+
+@flexitest.register
+class ElSelfDestructToAddressContractTest(testenv.StrataTester):
+    def __init__(self, ctx: flexitest.InitContext):
+        install_solc(version="0.8.16")
+        set_solc_version("0.8.16")
+        ctx.set_env("prover")
+
+    def main(self, ctx: flexitest.RunContext):
+        reth = ctx.get_service("reth")
+        reth_rpc = reth.create_rpc()
+
+        prover_client = ctx.get_service("prover_client")
+        prover_client_rpc = prover_client.create_rpc()
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+
+        # Deploy the contracts
+        # The setup is a simple two contacts relation:
+        #  - Delegator is a receiver of the SELFDESTRUCT opcode.
+        #  - Suicider's only purpose is to be constructed with delegator's address
+        #       and suicide itself.
+        # Later on, the Suicider is called to be destructed and reproduce the bug.
+
+        # STEP 1: deploy delegator and fetch its address
+        delegator_abi, delegator_bytecode = SmartContracts.compile_contract(
+            "Counter.sol", "Counter"
+        )
+        delegator_contract = web3.eth.contract(abi=delegator_abi, bytecode=delegator_bytecode)
+        delegator_tx_hash = delegator_contract.constructor().transact()
+        tx_receipt_delegator = web3.eth.wait_for_transaction_receipt(delegator_tx_hash, timeout=30)
```

Comment:
Can we shift this logic into a library?  It seems like the kind of thing that's very repetitive.

Not saying to do this now, but we could deploy it as part of env init so we don't have to do it in the tests, right?

Reply:
Yeah, agree. Created a separate [ticket](https://alpenlabs.atlassian.net/browse/STR-1461) for that, likely will address it in the next week.

If you have other examples or tickets in mind - please drop a comment there.